### PR TITLE
Distinguish different errors in case `borg check` failed.

### LINKED ISSUE #1143: Distinguish between data errors and network errors when running `borg check`
**Describe the bug**
If I click on "verify" to check the consistency of the backup, and the connection is interrupted, borg reports a damaged backup, but If i check the log it says that it simply cannot complete the check.

```
2021-12-19 11:34:51,361 - vorta.borg.borg_job - INFO - Remote: Starting repository index check
2021-12-19 11:34:51,361 - vorta.borg.borg_job - INFO - Remote: Index object count match.
2021-12-19 11:34:51,769 - vorta.borg.borg_job - INFO - Remote: Completed repository check, no problems found.
2021-12-19 11:34:51,769 - vorta.borg.borg_job - INFO - Starting archive consistency check...
2021-12-19 11:34:56,273 - vorta.borg.borg_job - INFO - Analyzing archive fedora-2021-10-30-111032 (1/15)
2021-12-19 11:35:37,700 - vorta.borg.borg_job - INFO - Analyzing archive fedora-2021-11-05-214200 (2/15)
2021-12-19 11:36:18,410 - vorta.borg.borg_job - INFO - Analyzing archive fedora-2021-11-12-124400 (3/15)
2021-12-19 11:37:00,527 - vorta.borg.borg_job - INFO - Analyzing archive fedora-2021-11-18-124159 (4/15)
2021-12-19 11:42:52,564 - vorta.borg.borg_job - WARNING - Remote: client_loop: send disconnect: Broken pipe
2021-12-19 11:42:52,705 - vorta.borg.borg_job - ERROR - Connection closed by remote host
```

**To Reproduce**
1. Go to 'Archive'
2. Click on 'Verify'
3. Suspend laptop
4. turn on again the laptop
5. you can see the warning to repair your backup

**Environment (please complete the following information):**
- OS: Fedora 35
- Vorta version: 0.8.2
- Installed from: flatpak

Vorta and Borg versions can be found in Main Window > Misc Tab.

**Additional context**
If appropriate include logs. Can be found in Main Window > Misc Tab > Log.

### PR DESCRIPTION
**Pull request by real-yfprojects**
In case the `BorgCheckJob` fails vorta switches over the returncodes and reacts accordingly.
Fixes #1143.

| returncode | meaning | behaviour |
---------------  | ------------| --------------|
0                 | success (logged as INFO) | ignored
1                 | warning (operation reached its normal end, but there were warnings – you should check the log, logged as WARNING) | `QMessageBox` with warning icon and text stating the warning message from borg
2                 | error (like a fatal error, a local or remote exception, the operation did not reach its normal end, logged as ERROR) | `QMessageBox` with error icon stating the error message from borg and `Consider repairing or recreating the repository soon to avoid missing data.`
128+N |killed by signal N (e.g. 137 == kill -9) | `QMessageBox` with error icon stating that the borg process was killed by signal N|

### CODE REVIEW

File: src/vorta/application.py
Comment (real-yfprojects) [inline src/vorta/application.py:23]: My IDE is using `isort` to sort imports. I didn't notice this found its way into a commit. Sorry about that.

Comment (m3nu) [reply src/vorta/application.py:23]: No worries. I plan to use Black and isort to clean up the whole code base at some point. It's just very destructive to the history and if distros want to cherry-pick only certain commits.

**Comment by m3nu** [comment]
Looks good and well thought-out at first glance. Will run it for a while tomorrow and then merge. Thanks for the great contribution!

**Comment by m3nu** [comment]
Minor linting:

```
./src/vorta/application.py:318:14: E261 at least two spaces before inline comment
./src/vorta/application.py:321:51: E261 at least two spaces before inline comment
./src/vorta/borg/borg_job.py:217:28: E261 at least two spaces before inline comment
```

**Comment by m3nu** [comment]
Our status text can only show a string:

```
Traceback (most recent call last):
  File "/Users/manu/Workspace/vorta/src/vorta/application.py", line 348, in check_failed_response
    msg.setInformativeText(infotext)
TypeError: setInformativeText(self, str): argument 1 has unexpected type 'tuple'
```

(Happens when removing some random files from a repo and then running check on it.)

**Comment by bastiencyr** [comment]
For docstring, is there a point of convergence @m3nu ? @real-yfprojects uses numpydoc. Pycharm uses by default reST (also used by spinx), see https://stackoverflow.com/questions/3898572/what-are-the-most-common-python-docstring-formats

**Comment by real-yfprojects** [comment]
Numpydoc is supported by sphinx as well. 

It doesn't seem like a specific doc style is already established in vorta but it might make sense to agree on one. I am used to numpydoc but I am ok with reST too because my docstring generator supports it.

**Comment by m3nu** [comment]
Feel free to decide a style and we put it on the contribution page. None is established. Myself I usually just write a few sentences on the function but didn't feel the need to formally list arguments.

**Comment by m3nu** [comment]
+1 for Spinx/Google docstrings, if we pick one. More readable IMHO.

But again, I prefer clear naming or args and a short summary for internal functions. Good docstrings make more sense when other people will use your code as library.

(Both just personal opinions of course.)

File: src/vorta/application.py
Comment (real-yfprojects) [inline src/vorta/application.py:308]:
```suggestion
        error_message = errors[0][1] if errors else ''
```

This would revert last change.

This PR was way to error-prone. I will do better next time.

**Comment by real-yfprojects** [comment]
> Feel free to decide a style and we put it on the contribution page. None is established.

Then it is up to @bastiencyr to decide. Although being the default of PyCharm shouldn't be an argument since devs are capable of changing the default to *numpydoc*.

> But again, I prefer clear naming or args and a short summary for internal functions. Good docstrings make more sense when other people will use your code as library.

Clear naming should be a necessity. But when creating this PR, detailed docstrings that state what a function does, how it does it in general or what it outsources, could have been very helpful for finding where the new feature has to be implemented. For this one I managed without any docstrings but when I was trying to implement a feature for the *spyder* repo it was an actual pain.

**Comment by m3nu** [comment]
Confirming that this PR does what it promises. Tested error case, happy case and cancelled case.

**Review by m3nu — APPROVED** [review]

**Comment by m3nu** [comment]
Congrats on your first merged PR, @real-yfprojects! 🎉🤝

**Comment by real-yfprojects** [comment]
Well, thanks!
But your last commit is very irritating. I just used the suggestion tool to explain my last force push. I think there was a misunderstanding.

**Comment by m3nu** [comment]
> But your last commit is very irritating. I just used the suggestion tool to explain my last force push. I think there was a misunderstanding.

This was your suggested change here on Github. Also couldn't pull after it. Weird stuff. It did avoid passing a tuple and passed the right error message.

**Comment by real-yfprojects** [comment]
> This was your suggested change here on Github. Also couldn't pull after it. Weird stuff. It did avoid passing a tuple and passed the right error message.

A pull after a force push will always lead to a merge. Instead you have to force pull with `git pull -f`. This is because a force push means that the commit history was rewritten to make it clearer which is ok as long you are the only one committing to the branch.

The error message now won't be correct in case borg logged a warning before the error.

**Comment by m3nu** [comment]
I did it over the web UI only.

> The error message now won't be correct in case borg logged a warning before the error.

Care to do a small PR for that?

**Comment by real-yfprojects** [comment]
Yes, let's see whether I can use the same branch and this PR gets updated.

**Comment by real-yfprojects** [comment]
> Yes, let's see whether I can use the same branch and this PR gets updated.

It doesn't work this way. I will create a new PR.

**Pull request merged**
